SparkPost/matchbox
# Resolve issue #214 to incorporate modal component keyboard tab trap

Resolves #214 

## What Changed
- Incorporated [React Focus Lock NPM package](https://www.npmjs.com/package/react-focus-lock)
- Incorporated Cypress end to end test runner
- Incorporated Cypress tab plugin to allow testing of user interaction using `tab` and `shift + tab`
- Updated modal `<Content/>` component to receive focus on mount *and* to leverage the `<FocusLock/>` component
- Incorporated end to end testing in the CI build which included:
    - Incorporated unit test and end to end testing in to a single test command
    - Added `http-server` NPM package to run "production" build of Storybook UI in the CI server
    - Added `start-server-and-test` to allow Cypress to wait for `http-server` to launch before running end to end testing
- Updated failing snapshots where necessary

## How to Test
- Run Cypress locally to verify the tab trap occurs as expected
- Visit http://localhost:9001/?selectedKind=Overlays%7CModal&selectedStory=Toggle%20Example&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel and verify that when the modal opens, focus moves to the modal (using the console and `document.activeElement`). Then, use `tab` and `shift + tab` to move within the modal without moving outside of the modal
- After closing the modal, resume tabbing through the DOM as usual
